fastapi/encoders.py

```diff
@@ -105,24 +105,22 @@ def jsonable_encoder(
                 )
             )
         return encoded_list
+    for encoder_type, encoder in custom_encoder.items():
+        if isinstance(obj, encoder_type):
+            return encoder(obj)
+    for encoder_type, encoder in ENCODERS_BY_TYPE.items():
+        if isinstance(obj, encoder_type):
+            return encoder(obj)
     errors: List[Exception] = []
     try:
-        if custom_encoder and type(obj) in custom_encoder:
-            encoder = custom_encoder[type(obj)]
-        else:
-            encoder = ENCODERS_BY_TYPE[type(obj)]
```

Comment:
This is *much* more performant than the
```python
    for encoder_type, encoder in ENCODERS_BY_TYPE.items():
        if isinstance(obj, encoder_type):
            return encoder(obj)
```
approach you've replaced this with.

I think for performance, you should keep this check as is, and just add the `isinstance` checks afterward in case of failure. That will keep existing code fast (as long as it doesn't raise errors).